### DISCUSSION TITLE
How to ignore non-existing dims given in chunks?

### DISCUSSION THREAD

**Discussion by willirath** [pydata/xarray#8043] [Q&A]
Is there a way of over-specifying chunks upon opening a dataset without throwing an error?

Currently, giving chunk sizes along dimensions that are not present in the dataset fails with a `ValueError`:
```python
da = xr.DataArray(np.arange(10), dims=("x", ), name="example_data")
da.to_dataset().to_netcdf("example.nc")
ds = xr.open_dataset("example.nc", chunks={"x": 5, "y": 1})
```

**Comment by dcherian** [comment]
I don't think so. You can do `chunks=5` which will chunk along all dimensions.

**Comment by TomNicholas** [comment]
@willirath if the problem is that you don't know the dimensions of the dataset until after you've opened it, you could always open first then chunk afterwards, I think the result is the same:
```python
chunks = {'x': 5}

ds = xr.open_dataset("example.nc")

if 'y' in ds.dims:
    chunks['y'] = 1

ds = ds.chunk(chunks=chunks)
```
